swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -188,7 +188,11 @@ func runFileRetrievalTest(nodeCount int) error {
 		if err != nil {
 			return err
 		}
-		if _, err := sim.WaitTillHealthy(ctx); err != nil {
+		snap, err := simulation.ReadSnapshot(filename)
```

Comment:
No i really meant that this should be internal to `sim.UploadSnapshot` and it should block until "recreated".